src/server/services/database.service.ts

```diff
@@ -55,4 +63,33 @@ export class DatabaseService {
       );
     });
   }
+
+  async writePost(userPost: DatabasePost): Promise<DatabasePost> {
+    const { userId, title, body, image } = userPost;
+
+    const queryResult = await this.pool.connect(async (connection) => {
+      return connection.one<DatabasePost>(
+        sql`INSERT INTO posts (user_id, title, body, image) VALUES (${userId}, ${title}, ${body}, ${image}) RETURNING title, body, image`
+      );
+    });
+    return queryResult;
+  }
+
+  async deletePost(postId: string, userId: string) {
```

Comment:
since all your others have return type, good to keep it consistent and have one here too